scripts/update-bootnodes.sh

```diff
@@ -0,0 +1,330 @@
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+log_info() {
+    echo -e "${GREEN}[INFO]${NC} $*"
+}
+
+log_warn() {
+    echo -e "${YELLOW}[WARN]${NC} $*"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $*"
+}
+
+# Function to fetch bootnodes from etcnodes API
+fetch_bootnodes_from_sources() {
+    log_info "Fetching bootnodes from etcnodes API..."
+    
+    # Fetch from etcnodes API - the authoritative source of live ETC nodes
+    local ETCNODES_API="https://api.etcnodes.org/peers"
+    
+    log_info "Fetching from ${ETCNODES_API}..."
+    
+    # Fetch and extract enode URLs from JSON response
+    # Get all enodes, we'll prioritize standard port nodes during selection
+    curl -s "${ETCNODES_API}" | \
+        jq -r '.[].enode' 2>/dev/null | \
+        grep -o 'enode://[^"]*' | \
+        sort -u > "${TEMP_DIR}/etcnodes_bootnodes.txt"
+    
+    local etcnodes_count=$(wc -l < "${TEMP_DIR}/etcnodes_bootnodes.txt" || echo 0)
+    log_info "Found ${etcnodes_count} live bootnodes from etcnodes API"
+    
+    # Use etcnodes as the external source
+    cp "${TEMP_DIR}/etcnodes_bootnodes.txt" "${TEMP_DIR}/all_external_bootnodes.txt"
+    
+    # Also add current bootnodes to available pool (they might still be valid)
+    # Make sure current_bootnodes.txt exists before trying to cat it
+    if [ -f "${TEMP_DIR}/current_bootnodes.txt" ]; then
+        cat "${TEMP_DIR}/current_bootnodes.txt" \
+            "${TEMP_DIR}/all_external_bootnodes.txt" 2>/dev/null | \
+            sort -u > "${TEMP_DIR}/available_bootnodes.txt"
+    else
+        cp "${TEMP_DIR}/all_external_bootnodes.txt" "${TEMP_DIR}/available_bootnodes.txt"
+    fi
+    
+    local count=$(wc -l < "${TEMP_DIR}/available_bootnodes.txt")
+    log_info "Total unique bootnodes from all sources: ${count}"
+}
+
+# Function to extract current bootnodes from config
+extract_current_bootnodes() {
+    log_info "Extracting current bootnodes from ${CONFIG_FILE}..."
+    
+    # Extract enodes from the bootstrap-nodes array
+    # Use awk to find start and end of the array reliably
+    awk '/bootstrap-nodes = \[/,/^\s*\]/' "${CONFIG_FILE}" | \
+        grep 'enode://' | \
+        grep -o 'enode://[^"]*' | \
+        sort -u > "${TEMP_DIR}/current_bootnodes.txt"
+    
+    local count=$(wc -l < "${TEMP_DIR}/current_bootnodes.txt")
+    log_info "Current configuration has ${count} bootnodes"
+}
+
+# Function to validate enode URL format
+validate_enode() {
+    local enode="$1"
+    
+    # Basic validation: should start with enode:// and have @ and :
+    # Node IDs are 128 hex characters (can be lowercase or uppercase)
+    if [[ "$enode" =~ ^enode://[a-fA-F0-9]{128}@[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+:[0-9]+ ]]; then
+        return 0
+    elif [[ "$enode" =~ ^enode://[a-fA-F0-9]{128}@[^:]+:[0-9]+(\?discport=[0-9]+)?$ ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Function to check if bootnode uses standard port (30303)
+has_standard_port() {
```

Comment:
the enodes received from the api will all have the port used for p2p with the api host. all enodes should be set to 30303 in the boot nodes